src/vs/workbench/contrib/inlineChat/browser/inlineChatController.ts

```diff
@@ -234,6 +244,64 @@ export class InlineChatController implements IEditorContribution {
 		return INLINE_CHAT_ID;
 	}
 
+	private _installProviderEnablementListeners() {
+		this._store.add(this._inlineChatService.onDidChangeProviders((e) => {
+			if (e.added) {
+				this._providerEnablementEvents.set(e.added, e.added.onDidChangeEnablementStatus(async () => {
+					if (this._enabledProvider === e.added) {
+						if (this._session?.provider === e.added) {
+							await this.cancelSession();
+						}
+						this._enabledProvider = undefined;
+					}
+					// TODO @lramos15 why can e.added be undefined
```

Comment:
@jrieken Any ideas

Reply:
`added` is undefined when removing a provider and vice versa

Reply:
Sure but this in a callback of `e.added` specifically `e.added.onDidChange....` so I guess I'm worried that it's not properly catching I'm referring to the `e.added` which fired the event